mindmaps-graql/src/main/java/io/mindmaps/graql/internal/analytics/Analytics.java

```diff
@@ -110,11 +115,19 @@ public Analytics(String keySpace) {
      */
     public Analytics(String keySpace, Set<Type> types) {
         this.keySpace = keySpace;
+        MindmapsGraph graph = MindmapsClient.getGraph(this.keySpace);
+
+        // collect resource-types for statistics
+        graph.getMetaResourceType().instances()
+                .forEach(type ->
+                        resourceTypes.put(type.getId(), type.asResourceType().getDataType().getName()));
 
         // use ako relations to add subtypes of the provided types
         for (Type t : types) {
             t.subTypes().forEach(subtype -> allTypes.add(subtype.getId()));
         }
+
+        graph.rollback();
```

Comment:
We should remove all those graph.rollback scattered during troubleshooting the problem of sharing MM objects b/w transactions.